cogs/poll.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord.py requirements
+import discord
+from discord.ext import commands
+import asyncio
+
+# for the alphabet list
+import string
+
+# for the timeout
+import time
+
+ALPHABET = list(string.ascii_uppercase)
+EMOJI_ALPHABET = [
+    u"\U0001F1E6", u"\U0001F1E7", u"\U0001F1E8", u"\U0001F1E9", u"\U0001F1EA",
+    u"\U0001F1EB", u"\U0001F1EC", u"\U0001F1ED", u"\U0001F1EE", u"\U0001F1EF",
+    u"\U0001F1F0", u"\U0001F1F1", u"\U0001F1F2", u"\U0001F1F3", u"\U0001F1F4",
+    u"\U0001F1F5", u"\U0001F1F6", u"\U0001F1F7"
+]    # :regional_indicator_a: to :regional_indicator_r:
+TIME_UNITS = {
+    "second": 1,
+    "seconds": 1,
+    "minute": 60,
+    "minutes": 60,
+    "hour": 60 * 60,
+    "hours": 60 * 60,
+    "day": 24 * 60 * 60,
+    "days": 24 * 60 * 60
+}
+
+
+class Poll:
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def poll(self, ctx, *args):
+        """
+        Creates a poll with a maximum of 18 options.
+        ==Examples==
+        ?poll "Question" "Option 1" "Option 2" "..."
+        ?poll -t 30 minutes "Question" "Option 1" "..."

```

Comment:
Instead of having "question" and -t which denotes the beginning of different input types, try to make them more consistent ("question" could be changed to -q for example), and I think there should be a better way to parse arguments

Reply:
I'm not sure I understand, question doesn't denote the beginning of an input type. The way it works is that the first argument that is not a switch, is the question, like this:
?poll "What do you think of that?" "Yes it's great" "No it's bad" 
And anywhere in this you could write -t 30 minutes

Reply:
ok yeah I misunderstood this part

Reply:
I will still make the examples clearer in case